downloadDirectory/src/main/java/org/reactome/release/downloadDirectory/OutputCuratedComplexInstances.java

```diff
@@ -0,0 +1,98 @@
+		
+		// Sort by display name
+		ArrayList<String> complexDisplayNames = new ArrayList<String>();
+		HashMap<String,GKInstance> sortedComplexInstances = new HashMap<String,GKInstance>();
+		for (GKInstance complexInst : complexInstances) {
+			complexDisplayNames.add((complexInst.getAttributeValue(ReactomeJavaConstants._displayName).toString()));
+			sortedComplexInstances.put(complexInst.getAttributeValue(ReactomeJavaConstants._displayName).toString(), complexInst);
+		}
+		Collections.sort(complexDisplayNames);
+		
+		// There are two output files for this module 'curated_complexes.txt' and 'curated_complexes.stid.txt'
+		// The only difference between them is the addition of a stable id column for the second file.
+		// Files are generated by populating ArrayLists with the lines of the file
+		ArrayList<String> complexLines = new ArrayList<String>();
+		ArrayList<String> stidComplexLines = new ArrayList<String>();
+		
+		// TODO: Small file differences (GKB::Utils#13-48-1362)
+		
+		// Add headers as first line
+		complexLines.add("#DB_ID\t#species.name[0]\t#_displayName\n");
+		stidComplexLines.add("#DB_ID\t#species.name[0]\t#_displayName\n");
+		
+		// Iterate through all retrieved instances
+		for (String displayName : complexDisplayNames) {
+			GKInstance complexInst = sortedComplexInstances.get(displayName);		
+			String complexLine = "";
+			String stidComplexLine = "";
+			
+			// Add stable id as first column of stable id file
+			stidComplexLine += ((GKInstance) complexInst.getAttributeValue(ReactomeJavaConstants.stableIdentifier)).getDisplayName() + "\t";
+			// Add DB ID
+			complexLine += complexInst.getDBID().toString() + "\t";
+			stidComplexLine += complexInst.getDBID().toString() + "\t";
+			
+			// Check that species attributes exist, and then add the display name to the line if it exists. Otherwise, just add a tab.
+			// TODO: Pipe separators for multi-species instances
+			if ((complexInst.getAttributeValue(ReactomeJavaConstants.species) != null && ((GKInstance) complexInst.getAttributeValue(ReactomeJavaConstants.species)).getDisplayName() != null)) {
+				Collection<GKInstance> speciesInstances = complexInst.getAttributeValuesList(ReactomeJavaConstants.species);
+				List<String> speciesNames = new ArrayList<String>();
+				for (GKInstance speciesInst : speciesInstances) {
+					speciesNames.add(speciesInst.getDisplayName());
+				}
+				complexLine += String.join("|", speciesNames) + "\t";
+				stidComplexLine += String.join("|", speciesNames) + "\t";
+			} else {
+				complexLine += "\t";
+				stidComplexLine += "\t";
+			}
+			
+			// Final column for each file the Complexes display name
+			complexLine += complexInst.getDisplayName() + "\n";
+			stidComplexLine += complexInst.getDisplayName() + "\n";
+			
+			//Add completed line to associated array
+			complexLines.add(complexLine);
+			stidComplexLines.add(stidComplexLine);
+		}
```

Comment:
It might be worth generating small named methods to return the values you need to build the complex and stidComplex output lines.  Then you could build them in one line with calls to the methods so the content added is readable from left to right.

Reply:
Also, can wait until after the release